graphdatascience/query_runner/neo4j_query_runner.py

```diff
@@ -124,10 +133,19 @@ def run_cypher(
             else:
                 self._last_bookmarks = session.last_bookmarks()
 
-            notifications = result.consume().notifications
-            if notifications:
-                for notification in notifications:
-                    self._forward_cypher_warnings(notification)
+            if (
+                Neo4jQueryRunner._NEO4J_DRIVER_VERSION < ServerVersion(5, 21, 0)
+                or result._warn_notification_severity != "WARNING"
+            ):
+                notifications = result.consume().notifications
+                if notifications:
+                    for notification in notifications:
+                        self._forward_cypher_warnings(notification)
+            elif Neo4jQueryRunner._NEO4J_DRIVER_VERSION > ServerVersion(5, 21, 0):
+                # the client does not expose YIELD fields so we just skip these warnings for now
+                warnings.filterwarnings(
+                    "ignore", message=r".*The query used a deprecated field from a procedure\. .* by 'gds.* "
+                )
```

Comment:
I did think a while about what first this check was doing and eventually realized it is to say that every notification that isn't even a warning in the first place should be forwarded anyways. Maybe it's also the room temperature that it took a while to understand 😅 

I wonder if flipping the conditions would make it a bit easier to read, maybe:


```suggestion
            if (
                Neo4jQueryRunner._NEO4J_DRIVER_VERSION >= ServerVersion(5, 21, 0)
                and result._warn_notification_severity == "WARNING"
            ):
                # the client does not expose YIELD fields so we just skip these warnings for now
                warnings.filterwarnings(
                    "ignore", message=r".*The query used a deprecated field from a procedure\. .* by 'gds.* "
                )
            else:
                notifications = result.consume().notifications
                if notifications:
                    for notification in notifications:
                        self._forward_cypher_warnings(notification)
```

Reply:
I also confused myself while writing. Thanks for the idea :)